src/objective-c/GRPCClient/private/GRPCWrappedCall.m

```diff
@@ -0,0 +1,341 @@
+  return [self initWithHandler:nil];
+}
+
+- (instancetype) initWithHandler:(void (^)(NSError *))handler {
+  if (self = [super init]) {
+    _handler = handler;
+    _code = gpr_malloc(sizeof(grpc_status_code));
+    _details = gpr_malloc(sizeof(char*));
+    _details_capacity = gpr_malloc(sizeof(size_t));
+    *_details_capacity = 0;
+    _recv_trailing_metadata = gpr_malloc(sizeof(grpc_metadata_array));
+  }
+  return self;
+}
+
+- (void)getOp:(grpc_op *)op {
+  op->op = GRPC_OP_RECV_STATUS_ON_CLIENT;
+  op->data.recv_status_on_client.status = _code;
+  op->data.recv_status_on_client.status_details = _details;
+  op->data.recv_status_on_client.status_details_capacity = _details_capacity;
+  op->data.recv_status_on_client.trailing_metadata = _recv_trailing_metadata;
+}
+
+- (void (^)(void))opProcessor {
+  return ^{
+    grpc_status status;
+    status.status = *_code;
+    status.details = *_details;
+    status.metadata = _recv_trailing_metadata;
+    gpr_free(_code);
+    gpr_free(_details);
+    gpr_free(_details_capacity);
+    if (_handler) {
+      _handler([NSError grpc_errorFromStatus:&status]);
+    }
+  };
+}
+
+@end
+
+@implementation GRPCWrappedCall{
+  grpc_call *_call;
+  GRPCCompletionQueue *_queue;
+}
+
+- (instancetype)init {
+  return [self initWithChannel:nil method:nil host:nil];
+}
+
+- (instancetype)initWithChannel:(GRPCChannel *)channel
+                         method:(NSString *)method
+                           host:(NSString *)host {
+  if (!channel || !method || !host) {
+    [NSException raise:NSInvalidArgumentException
+                format:@"channel, method, and host cannot be nil."];
+  }
+  
+  if (self = [super init]) {
+    static dispatch_once_t initialization;
+    dispatch_once(&initialization, ^{
+      grpc_init();
+    });
+    
+    _queue = [GRPCCompletionQueue completionQueue];
+    _call = grpc_channel_create_call(channel.unmanagedChannel, _queue.unmanagedQueue,
+                                     method.UTF8String, host.UTF8String, gpr_inf_future);
+    if (_call == NULL) {
+      return nil;
+    }
+  }
+  return self;
+}
+
+- (void)startBatchWithOperations:(NSArray *)operations {
+  [self startBatchWithOperations:operations errorHandler:nil];
+}
+
+- (void)startBatchWithOperations:(NSArray *)operations errorHandler:(void (^)())errorHandler {
+  NSMutableArray *opProcessors = [NSMutableArray array];
+  size_t nops = operations.count;
+  grpc_op *ops_array = gpr_malloc(nops * sizeof(grpc_op));
+  size_t i = 0;
+  for (id op in operations) {
+    [op getOp:&ops_array[i]];
+    [opProcessors addObject:[op opProcessor]];
+  }
+  grpc_call_error error = grpc_call_start_batch(_call, ops_array, nops,
+                                                (__bridge_retained void *)(^(grpc_op_error error){
+    if (error != GRPC_OP_OK) {
+      if (errorHandler) {
+        errorHandler();
+      } else {
+        [NSException raise:@"Operation Exception" format:@"The batch failed with an unknown error"];
```

Comment:
That's what the error handler is for :) Let's make sure `GRPCCall` passes an error handler whenever it's significant.